roles/openshift_inventory/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+- no_log: True  # keep task descriptions legible
+  block:
+
+  - name: Standardize on latest variable names
+    set_fact:
+      # goal is to deprecate deployment_type in favor of openshift_deployment_type.
+      # both will be accepted for now, but code should refer to the new name.
+      # TODO: once this is well-documented, add deprecation notice if using old name.
+      deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+      openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
```

Comment:
I'm not super confident in my understanding of how this YAML-programming actually works without doing local tests, but here's the logic I'm following:

- `set_fact` sets a variable
- `deployment_type` should be a string (conceptually)
- `deployment_type: {{ openshift_deployment_type | default(deployment_type) }}` means set `deployment_type` to `openshift_deployment_type` or to the value of itself?! I don't know what happens if we access a variable that is undefined... I verified it locally, it fails with a rather hard to grasp message:
```
TASK [set_fact] ****************************************************************
fatal: [localhost]: FAILED! => {
    "failed": true
}

MSG:

the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'deployment_type' is undefined

The error appears to have been in '/home/vagrant/src/github.com/openshift/openshift-ansible/foo.yml': line 4, column 5, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
  - set_fact:
    ^ here
```

- So it seems that `default` allows us to reference undefined names without throwing errors...

My suggestion would be the empty string or no argument... but I don't know what different it makes. I tried all that in this test playbook and `default()`, `default(''`) and `default(None)` all print the same output!

```
$ cat foo.yml 
- hosts: localhost
  connection: local
  tasks:
  - set_fact:
      deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default() }}"
  - debug:
      msg: "deployment_type = {{ deployment_type }}"
[vagrant@localhost openshift-ansible]$ ansible-playbook -i hosts foo.yml 

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [set_fact] ****************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {}

MSG:

deployment_type = 


PLAY RECAP *********************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0  
```

---

So, a suggestion, perhaps we could do the same as we do below for `openshift_release` and mark it explicitly as a string?

```yaml
set_fact:
  openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default | string }}"
  deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default | string }}"
  openshift_release: "{{ openshift_release | default | string | regex_replace('^v', '') }}"
```